great_expectations/rule_based_profiler/parameter_builder/regex_pattern_string_parameter_builder.py

```diff
@@ -0,0 +1,152 @@
+            candidate_strings: a list of candidate date format strings that will REPLACE the default
+            additional_candidate_strings: a list of candidate date format strings that will SUPPLEMENT the default
+            data_context: DataContext
+            batch_request: specified in ParameterBuilder configuration to get Batch objects for parameter computation.
+        """
+
+        super().__init__(
+            name=name,
+            data_context=data_context,
+            batch_request=batch_request,
+        )
+        self._metric_domain_kwargs = metric_domain_kwargs
+        self._metric_value_kwargs = metric_value_kwargs
+
+        self._threshold = threshold
+        if candidate_strings is not None:
+            self._candidate_strings = set(candidate_strings)
+        else:
+            self._candidate_strings = (
+                RegexPatternStringParameterBuilder.CANDIDATE_STRINGS
+            )
+
+    def _build_parameters(
+        self,
+        parameter_container: ParameterContainer,
+        domain: Domain,
+        *,
+        variables: Optional[ParameterContainer] = None,
+        parameters: Optional[Dict[str, ParameterContainer]] = None,
+    ):
+        """
+        Check the percentage of values matching the REGEX string, and return the best fit, or None if no
+        string exceeds the configured threshold.
+
+        :return: ParameterContainer object that holds ParameterNode objects with attribute name-value pairs and optional details
+        """
+        validator: Validator = self.get_validator(
+            domain=domain, variables=variables, parameters=parameters
+        )
+
+        batch_ids: Optional[List[str]] = self.get_batch_ids(
+            domain=domain, variables=variables, parameters=parameters
+        )
+        if not batch_ids:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"Utilizing a {self.__class__.__name__} requires a non-empty list of batch identifiers."
+            )
+
+        nonnull_metrics: MetricComputationResult = self.get_metrics(
+            batch_ids=batch_ids,
+            validator=validator,
+            metric_name="column_values.nonnull.count",
+            metric_domain_kwargs=self._metric_domain_kwargs,
+            domain=domain,
+            variables=variables,
+            parameters=parameters,
+        )
+        nonnull_count: int = sum(nonnull_metrics.metric_values)
+
+        regex_string_success_ratios: dict = {}
+        for regex_string in self._candidate_strings:
+            if self._metric_value_kwargs:
+                match_regex_metric_value_kwargs: Tuple[dict] = (
+                    {
+                        **self._metric_value_kwargs,
+                        **{"regex": regex_string},
+                    },
+                )
+
+            else:
+                match_regex_metric_value_kwargs: dict = {"regex": regex_string}
```

Comment:
Is there a reason why we can't use a unified type here? Why is the `Tuple[dict]` necessary?

I only ask because if possible, I'd like to adhere to a single type to reduce variability. The more options we have, the more brittle/complex the code becomes.

Reply:
great point. This has been changed to just `dict`